Fix flaky test for UDP snmptrap

This test keeps failing. My guess is that the UDP packets are getting dropped.

```
    --- FAIL: TestServer_AlertHandlers/snmptrap (0.04s)
    	server_test.go:7779: snmptrap: unexpected snmptrap request:
    		exp
    		[{Pdu:{IsNil:false Type:SNMPTrapV2 ErrorStatus:NoError ErrorIndex:0 VarBinds:[{Oid:1.3.6.1.6.3.1.1.4.1.0 Value:1.1.2 Type:Oid} {Oid:1.1.2.1 Value:message Type:OctetString} {Oid:1.1.2.2 Value:CRITICAL Type:OctetString}]} Error:<nil>}]
    		got
    		[]
```

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -7543,6 +7543,7 @@ func TestServer_AlertHandlers(t *testing.T) {
 				c.SNMPTrap.Enabled = true
 				c.SNMPTrap.Addr = ts.Addr
 				c.SNMPTrap.Community = ts.Community
+				c.SNMPTrap.Retries = 3
 				return ctxt, nil
 			},
 			result: func(ctxt context.Context) error {

--- a/services/config/service_test.go
+++ b/services/config/service_test.go
@@ -1035,6 +1035,7 @@ func TestService_GetConfig(t *testing.T) {
 		basePath := server.Server.URL + httpd.BasePath + "/config"
 		// Apply all updates
 		for _, update := range tc.updates {
+			errC := make(chan error, 1)
 			go func() {
 				// Validate we got the update over the chan.
 				// This keeps the chan unblocked.
@@ -1043,8 +1044,9 @@ func TestService_GetConfig(t *testing.T) {
 				select {
 				case cu := <-updates:
 					cu.ErrC <- nil
+					errC <- nil
 				case <-timer.C:
-					t.Fatal("expected to get config update")
+					errC <- errors.New("expected to get config update")
 				}
 			}()
 			resp, err := http.Post(basePath+update.Path, "application/json", strings.NewReader(update.Body))
@@ -1058,6 +1060,14 @@ func TestService_GetConfig(t *testing.T) {
 			}
 			if got, exp := resp.StatusCode, http.StatusNoContent; got != exp {
 				t.Fatalf("update failed: got: %d exp: %d\nBody:\n%s", got, exp, string(body))
+			}
+			select {
+			case err := <-errC:
+				if err != nil {
+					t.Fatal(err)
+				}
+			default:
+				t.Fatal("expected to get an response on errC")
 			}
 		}
 

